### ISSUE TITLE
Move the function `reverse` from Data.Fin.Properties to Data.Fin

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#203]

**Comment by MatthewDaggitt** [comment]
Maybe not so `low-hanging-fruit`!

Just tried to do it by opening `Data.Nat.Properties` in `Data.Fin` and got:
```
cyclic module dependency:
  Data.Fin
  Data.Nat.Properties
  Algebra.RingSolver.Simple
  Algebra.RingSolver
  Data.Fin
```

**Comment by bch29** [comment]
`reverse` could go in the new `Data.Fin.Permutation` instead, since it does after all form a permutation.

**Comment by MatthewDaggitt** [comment]
> reverse could go in the new Data.Fin.Permutation instead, since it does after all form a permutation.

Yes, good idea!

**Comment by MatthewDaggitt** [comment]
Closed with e5e5d593937403b415e54bb886723fb2c70175a8
